docs/work-with-data-apis.md

```diff
@@ -0,0 +1,66 @@
+# Work with unsupported REST Data APIs
+
+HERE Data SDK for C++ supports only a limited number of REST Data APIs. You can find a list of them on our page with the [overall architecture](OverallArchitecture.md).
+
+If you need to work with the REST Data APIs that are currently not supported, you can configure the Data SDK with a unique base URL.
+
+The example below demonstrates how to use the Data SDK to call the API for downloading the difference between two versions of a versioned catalog.
+
+**Example. Download the difference between two versions of a catalog.**
+
+1. Create the `OlpClientSettings` object.
+
+   For instructions, see [Create platform client settings](create-platform-client-settings.md).
+
+2. Get a base URL from the lookup API service.
+
+    For instructions, see the [API Lookup API Developer's Guide](https://developer.here.com/documentation/api-lookup/dev_guide/index.html).
+
+    You can now use this URL as the base for all requests to the REST Data API.
+
+3. Create an `OlpClient` instance with the client settings from step 1 and base URL from step 2.
+
+    ```cpp
+    olp::client OlpClient client(client_settings, base_url);
+    ```
+
+4. Prepare the REST API request parameters and the metadata URL.
+
+    ```cpp
+    std::multimap<std::string, std::string> header_params;
+    header_params.emplace("Accept", "application/json");
+
+    std::string metadata_url = "/layers/my_layer/changes";
+
+    std::multimap<std::string, std::string> query_params;
+
+    // Required parameters.
+    query_params.emplace("startVersion", "0");
+    query_params.emplace("endVersion", "1");
+
+    // Optional parameters.
+    query_params.emplace("additionalFields", "dataSize,checksum");
+    query_params.emplace("billingTag", "Billing1234");
+
+    CancellationContext context;
+    olp::client::HttpResponse response = client.CallApi(
+        metadata_url, "GET", query_params, header_params, {}, nullptr, "", context);
+    ```
+
+5. Check the response and, if needed, parse JSON.
+
+    ```cpp
+    >if (response.GetStatus() != olp::http::HttpStatusCode::OK) {
+    }
+    ```
+
+6. If an error occurs, print "Error", and then exit.
+
+7. Get the response.
```

Comment:
I would write: If the above check passed and the status of the received `olp::client::HttpResponse` is olp::http::HttpStatusCode::OK, i.e. 200 OK, then you can proceed with extracting the encoded JSON from the response and parsing it by using RapidJSON or any other JSON parsing library.

```cpp
    std::string response_json;
    response.GetResponse(response_json);
```